File: backend/components/metiers/threed/api/samples/db/sample_db.py
# Exemple ultra avancé clé en main du mock DB API Threed (Python)
from ..db.db import db_find_by_id, db_insert, db_update, db_delete
from ..audit.audit import audit_entity
from ..rgpd.rgpd import rgpd_sanitize
from ..accessibility.accessibility import check_accessibility
import logging

def sample_db_ultra():
    # Création
    data = {'name': 'UltraDB', 'status': 'active'}
    created = db_insert('threed', data)
    audit_entity(created, 'create')
    logging.info(f"Création: {created}")

    # Lecture
    entity = db_find_by_id('threed', created['id'])
    entity = rgpd_sanitize(entity)
    check_accessibility(entity)
    logging.info(f"Lecture: {entity}")

    # Mise à jour
    updated = db_update('threed', created['id'], {'status': 'inactive'})
    audit_entity(updated, 'update')
    logging.info(f"Mise à jour: {updated}")

    # Suppression
    deleted = db_delete('threed', created['id'])
    audit_entity({'id': created['id']}, 'delete')
    logging.info(f"Suppression: {deleted}")

    # Edge case
    try:
        db_find_by_id('threed', -1)
    except Exception as e:
        logging.warning(f"Erreur accès: {e}")
    print('DB ultra avancé exécuté avec succès.')

File: backend/components/metiers/threed/api/samples/db/sample_db.js
/* global console */
// Exemple ultra avancé clé en main du mock DB API Threed (JS)
const db = require('../db/db');
const { auditEntity } = require('../audit/audit');
const { rgpdSanitize } = require('../rgpd/rgpd');
const { checkAccessibility } = require('../accessibility/accessibility');

async function sampleDbUltra() {
  // Création
  const data = { name: 'UltraDB', status: 'active' };
  const created = db.insert('threed', data);
  auditEntity(created, 'create');
  console.info('Création:', created);

  // Lecture
  let entity = db.findById('threed', created.id);
  entity = rgpdSanitize(entity);
  checkAccessibility(entity);
  console.info('Lecture:', entity);

  // Mise à jour
  const updated = db.update('threed', created.id, { status: 'inactive' });
  auditEntity(updated, 'update');
  console.info('Mise à jour:', updated);

  // Suppression
  const deleted = db.delete('threed', created.id);
  auditEntity({ id: created.id }, 'delete');
  console.info('Suppression:', deleted);

  // Edge case
  try {
    db.findById('threed', -1);
  } catch (e) {
    console.warn('Erreur accès:', e);
  }
  console.log('DB ultra avancé exécuté avec succès.');
}

module.exports = { sampleDbUltra };
